### DISCUSSION TITLE
H1-G2 unable to read registers (which should be fine)

### DISCUSSION THREAD

**Discussion by sma0071** [nathanmarlor/foxess_modbus#594] [Support] [answered]
Hi, 
I have setup the EW11 and followed the wiring guide. Checked my cables and all seems to be connected corectly but my error is as follows for all enterays in HA.  

Inverter H1-6.0-E-G2

Any ideas?
tcp://192.168.1.10:502 247 - 5 failed poll attempts: now not connected. Last error: Error reading registers. Type: RegisterType.INPUT; start: 10016; count: 3; slave: 247 from tcp://192.168.1.10:502: Exception Response(132, 4, IllegalAddress)
tcp://192.168.1.10:502 247 - 5 failed poll attempts: now not connected. Last error: Error reading registers. Type: RegisterType.INPUT; start: 10016; count: 3; slave: 247 from tcp://192.168.1.10:502: Modbus Error: [Input/Output] Modbus Error: [Invalid Message] No response received, expected at least 8 bytes (0 received)

**Comment by canton7** [comment]
What version of foxess_modbus are you using?

**Comment by sma0071** [comment]
Currently I have v1.10.2b5 but I did also try going down to 1.9.0 as well. all seem to have the same error.

**Comment by sma0071** [comment]
Ok now I am using version v1.10.2b5 I am getting the following error:
Error connecting to inverter (friendly name: {friendly_name}). Error reading registers. Type: RegisterType.INPUT; start: 11000; count: 93; slave: 247 from tcp://192.168.1.10:502: Exception Response(132, 4, IllegalAddress)

**Reply by canton7** [reply]
I'm going to bring this over to #553

**Reply by canton7** [reply]
Can you post your full log when trying to go through the config flow?

**Reply by sma0071** [reply]
[home-assistant_2024-04-15T19-01-54.799Z.log](https://github.com/nathanmarlor/foxess_modbus/files/14983285/home-assistant_2024-04-15T19-01-54.799Z.log)

**Reply by sma0071** [reply]
I restarted and setup a new config. sorry just seen your reply.

**Reply by canton7** [reply]
It looks like your inverter isn't responding at all now?

**Reply by sma0071** [reply]
This is a new setup so I have not yet had it working with a modbus. It is still working with the cloud. I just enabled debug and the logs dont show much. Just the same communication error. Exception Response(131, 3, IllegalAddress)
And the following is now getting flagged as an error from HA. 
Connection Error
Error connecting to inverter (friendly name: {friendly_name}). Error reading registers. Type: RegisterType.HOLDING; start: 36001; count: 3; slave: 247 from tcp://192.168.1.10:502: Exception Response(131, 3, IllegalAddress)

[home-assistant_foxess_modbus_2024-04-15T19-13-12.139Z.log](https://github.com/nathanmarlor/foxess_modbus/files/14983394/home-assistant_foxess_modbus_2024-04-15T19-13-12.139Z.log)

**Reply by sma0071** [reply]
any ideas?

**Reply by canton7** [reply]
I'm confused. That latest log has auto-detection (during the config flow) failing to connect to your inverter:

```
2024-04-15 20:01:14.878 ERROR (MainThread) [custom_components.foxess_modbus.modbus_controller] Autodetect: failed to connect to (tcp://192.168.1.10:502)
```

Then there's a gap of 7 minutes, and suddenly it's detected the type of inverter, and is happily polling at least some of the registers. 

I'm just trying to find out what inverter model it thinks you have, but that information seems to have been cut out of your log.

**Reply by sma0071** [reply]
Is there any way I can enable a higher level of debug for you? I have not removed anything from the logs.

**Reply by sma0071** [reply]
[home-assistant_2024-04-15T19-35-11.207Z.log](https://github.com/nathanmarlor/foxess_modbus/files/14983557/home-assistant_2024-04-15T19-35-11.207Z.log)

I turned debug back on and hit confugre again. I think it has logged the type of inverter now.

**Reply by canton7** [reply]
It would really help to know what you're doing and seeing, and how that corresponds to the logs?

So far, I've seen:

1. Error messages which indicate that foxess_modbus thinks your inverter is a H1-G1, not a G2
2. Error messages which indicate that your inverter isn't responding at all
3. Error messages which indicate we're having trouble reading some fairly basic registers
4. Error messages which show it failing to get through the config flow, and then suddenly having succeeded in getting through the config flow

It's really hard to try and work out what you're doing which might be causing each of those.

**Reply by canton7** [reply]
Right, so it's definitely a G2, thanks. I've no idea why it thought it was a G1 in some of the logs.

So it's succeeding in reading 31007-31028 and 32000-32023, but it's failing to read 36001-36003.

@TonyM1958 any ideas?

**Reply by sma0071** [reply]
I have done the following, reload the service to try and get it to poll the inverter, delete and setup the module again ( I think this is why the communicaation errors happened). Turn on debug mode and then configure the module in HA again.
I am seeign this in the device page:
![image](https://github.com/nathanmarlor/foxess_modbus/assets/136631792/0dca7def-8c52-42db-8742-218f3b9c66ca)
![image](https://github.com/nathanmarlor/foxess_modbus/assets/136631792/8c86157e-74c1-43a8-bf9a-109fe4b45677)


and seeing this error in setting of HA
![image](https://github.com/nathanmarlor/foxess_modbus/assets/136631792/34a16c9b-87b8-4a1f-ae06-647c3fe8d863)

**Reply by TonyM1958** [reply]
36001 - 36003 are the Master, Slave and Manager version numbers that read OK on Jim's inverter. What are the firmware versions on this specific inverter shown in the app? Just wondering if there are different modbus mappings based on firmware...

Looking at the model number read from 30000, looks like it's saying 'H1-6.0-E-G2 - AUX' instead of H1-6.0-E-G2'?

**Reply by TonyM1958** [reply]
long shot, but one other thing to try might be reading 36001, 36002, 36003 indiviudally, just to see if it's failing to read 36001 as a block of 3? You could just do this in modpoll and report the values you get back?

**Reply by TonyM1958** [reply]
The firmware version registers on H1 are 30016, 30017, 30018 so it's probably worth checking if any values are being returned here as well?

The firmware we tested against were: Master 1.21 , Slave 1.02, Manager 1.30. I think we are now up to Manager 1.32.

**Reply by sma0071** [reply]
![Screenshot_2024-04-16-05-01-30-85_8e6f52239f4dc04be92d9657a7249325](https://github.com/nathanmarlor/foxess_modbus/assets/136631792/5611a0cf-f294-4852-88ab-c024031b32de)

**Reply by sma0071** [reply]
Sorry but I wouldn't know how to even go about trying to read the different blocks, if you need me to do that could you let me know how?

**Reply by TonyM1958** [reply]
The firmware is quite a long way behind what we have tested and Fox has made quite a few changes. Please can you ask your installer or Fox if they can update you to Master 1.21 and Manager 1.30 or later?

**Reply by sma0071** [reply]
Ok thanks, I will give it a go.

**Reply by canton7** [reply]
Hi, any update?

**Reply by sma0071** [reply]
Unfortunately not, my installer still hasn't returned my call requesting the firmware update. Do you know if going direct to fox would be quicker?

**Reply by FozzieUK** [reply, marked as answer]
@sma0071 yes if your installer doesn't respond, email Foxess service with your inverter serial number and ask for the inverter firmware to be updated they are usually pretty quick (but Monday to Friday only) - `service.uk@fox-ess.com`

**Reply by sma0071** [reply]
I have sent foxess the request. I will let you know as soon as I get an update. Thank you both for being so patient and helpful.

**Reply by sma0071** [reply]
I have had my firmware updated and everything seems to work perfectly now. Thanks for all your help.
![Screenshot_2024-04-22-16-23-33-13_c3a231c25ed346e59462e84656a70e50](https://github.com/nathanmarlor/foxess_modbus/assets/136631792/1bb8febf-6596-4a4c-bb1d-45e4076290bf)

**Reply by TonyM1958** [reply]
Good news. Thanks for letting us know.

**Reply by 60milesmile** [reply]
> @sma0071 yes if your installer doesn't respond, email Foxess service with your inverter serial number and ask for the inverter firmware to be updated they are usually pretty quick (but Monday to Friday only) - `service.uk@fox-ess.com`

I just posted a similar issue with a H1 G2 inverter - will get onto Foxxess customer services. thanks for this.
